src/main/java/com/powsybl/caseserver/service/FsCaseService.java

```diff
@@ -129,6 +131,23 @@ public String getCaseName(UUID caseUuid) {
         return caseInfos.getName();
     }
 
+    @Override
+    public String getDownloadCaseName(UUID caseUuid) {
+        String name = getCaseName(caseUuid);
+        if (Boolean.FALSE.equals(isTheFileOriginallyGzipped(caseUuid))) {
+            name = removeExtension(name, GZIP_EXTENSION);
+        }
+        return name;
+    }
+
+    @Override
+    public Boolean isTheFileOriginallyGzipped(UUID caseUuid) {
+        CaseMetadataEntity caseMetadata = getCaseMetaDataEntity(caseUuid);
+        String name = caseMetadata.getOriginalFilename();
+        String compressionFormat = caseMetadata.getCompressionFormat();
+        return !name.endsWith(GZIP_EXTENSION) || compressionFormat.equals(GZIP_FORMAT);
```

Comment:
Do not code it again. Use isCompressedCaseFile(caseName) 
Or  COMPRESSION_FORMATS.contains(compressionFormat)